python/tests/expressions/test_visitors.py

```diff
@@ -1024,7 +1024,7 @@ def test_missing_stats(schema: Schema, manifest_no_stats: ManifestFile) -> None:
         ), f"Should read when missing stats for expr: {expr}"
 
 
-def test_not(schema: Schema, manifest: ManifestFile) -> None:
+def test_not_nan(schema: Schema, manifest: ManifestFile) -> None:
```

Comment:
@Fokko, it looks like these renames were accidental. This is still testing `Not` expressions. I think the suffix `_nan` was added to a lot of these method names unintentionally.

Reply:
I replicated two suites to Python:
- [TestInclusiveMetricsEvaluator.java](https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/expressions/TestInclusiveMetricsEvaluator.java)
- [TestMetricsEvaluatorsNaNHandling.java](https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/expressions/TestMetricsEvaluatorsNaNHandling.java)

The ones coming from `TestMetricsEvaluatorsNaNHandling` have a `_nan` postfix to avoid duplicate naming. But I think we can just combine them.

Reply:
> The ones coming from TestMetricsEvaluatorsNaNHandling have a _nan postfix to avoid duplicate naming. But I think we can just combine them.

I don't understand why these are NaN tests though. They are identical to the way they were before and don't test anything specific to NaN. Am I missing something?